src/Common/ZooKeeper/ZooKeeperCachingGetter.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <Common/ZooKeeper/Common.h>
+
+
+namespace zkutil
+{
+
+class ZooKeeperCachingGetter : boost::noncopyable
+{
+public:
+    explicit ZooKeeperCachingGetter(zkutil::GetZooKeeper get_zookeeper_);
+
+    ZooKeeperCachingGetter(const ZooKeeperCachingGetter &) = delete;
+    ZooKeeperCachingGetter & operator=(const ZooKeeperCachingGetter &) = delete;
+
+    /// Returns the ZooKeeper session and the flag whether it was taken from the cache(false) or opened new(true),
+    /// because the session has expired or the cache was empty
+    std::pair<zkutil::ZooKeeperPtr, bool> getZooKeeper();
```

Comment:
`enum` instead of `bool` here could look better

Reply:
Reworked to use `enum`